cxx-qt-gen/src/parser/signals.rs

```diff
@@ -0,0 +1,145 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::parser::parameter::ParserParameter;
+use crate::syntax::fields::fields_named_to_ident_type;
+use syn::{Ident, ItemEnum, Result};
+
+/// Describes an individual Signal
+pub struct ParsedSignal {
+    /// The name of the signal
+    pub ident: Ident,
+    /// The parameters of the signal
+    pub parameters: Vec<ParserParameter>,
+    // TODO: later this will describe of the signal has an attribute
+    // stating that the signal exists in the base class
+}
+
+/// Describes a Signals block for a QObject
```

Comment:
```suggestion
/// Describes a Signals enum for a QObject
```

Reply:
Also, maybe rename `ParsedSignals` to `ParsedSignalsEnum` :thinking: 

It's easy to mix up with `ParsedSignal`